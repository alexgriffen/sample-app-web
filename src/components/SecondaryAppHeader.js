import React, { Component } from 'react';
import * as PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import { withStyles } from '@material-ui/core';
import { colors } from '../utils/Colors';
import headerBotPng from '../assets/img/headerBot2x.png';
import headerBotSvg from '../assets/svg/headerBot3x.svg';
import { FONT_FAMILY } from '../utils/Constants';

const styles = theme => ({
    root: {
      backgroundColor: colors.gray,
      boxShadow: 'none',
      height: 100,
      [ theme.breakpoints.down('md') ]: {
        height: 60,
      },
    },
    toolBar: {
      flex: 1,
      flexDirection: 'row',
      [ theme.breakpoints.down('md') ]: {
        paddingLeft: 10,
        paddingRight: 10,
      },
    },
    title: {
      color: colors.white,
      fontFamily: FONT_FAMILY,
      fontSize: 24,
      marginLeft: 16,
      textTransform: 'uppercase',
      [ theme.breakpoints.down('md') ]: {
        fontSize: 20,
        marginLeft: 0,
      },
    },
    headerBot: {
      marginLeft: 40,
      marginTop: 33,
      [ theme.breakpoints.down('md') ]: {
        display: 'none',
      },
    },
    leftComponent: {
      [ theme.breakpoints.down('md') ]: {
        width: '100%',
      },
    },
    rightComponent: {
      marginLeft: 'auto',
      marginRight: 16,
      [ theme.breakpoints.down('md') ]: {
        marginRight: 0,
      },
    },
  }
);
const { bool, func, string, object } = PropTypes;

class SecondaryAppHeader extends Component {
  constructor(props) {
    super(props);
  }

  static propTypes = {
    classes: object.isRequired,
    title: string,
    headerBot: bool,
    leftComponent: func,
    rightComponent: func,
  };

  static defaultProps = {
    title: null,
    headerBot: null,
    leftComponent: null,
    rightComponent: null,
  }

  /**
   * The title component in the header
   *
   * @return {element|null}
   */
  title() {
    const { classes, title } = this.props;

    return !title || (
      <span className={ classes.title }>{ title }</span>
    );
  }

  /**
   * The header bot in the header
   *
   * @return {element|null}
   */
  headerBot() {
    const { classes, headerBot } = this.props;

    return !headerBot || (
      <img className={ classes.headerBot } src={ headerBotPng } srcSet={ headerBotSvg }/>
    );

  }

  /**
   * The left component in the header
   *
   * @return {element|null}
   */
  leftComponent() {
    const { classes, leftComponent } = this.props;

    return !leftComponent || (
      <div className={ classes.leftComponent }>
        { React.cloneElement(leftComponent) }
      </div>
    );
  }

  /**
   * The right component in the header
   *
   * @return {element|null}
   */
  rightComponent() {
    const { classes, rightComponent } = this.props;

    return rightComponent ? (
        <div className={ classes.rightComponent }>
          { React.cloneElement(rightComponent) }
        </div>
      )
      : null;
  }

  render() {
    const { classes } = this.props;

    return (
      <AppBar className={ classes.root }>
        <Toolbar className={ classes.toolBar }>
          { this.title() }
          { this.headerBot() }
          { this.leftComponent() }
          { this.rightComponent() }
        </Toolbar>
      </AppBar>
    );
  }
}

export default withStyles(styles)(SecondaryAppHeader);
