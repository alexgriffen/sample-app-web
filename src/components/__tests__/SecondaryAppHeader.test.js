import React from 'react';
import { shallow } from 'enzyme/build';
import SecondaryAppHeader from '../SecondaryAppHeader';

describe('Secondary App Header', () => {
  it('should render the component correctly', () => {
    const component = shallow(
      <SecondaryAppHeader/>
    ).dive();

    expect(component).toMatchSnapshot();
  });

  it('should render a full option component correctly', () => {
    const component = shallow(
      <SecondaryAppHeader
        title="Full options"
        headerBot
        leftComponent={ <div/> }
        rightComponent={ <div/> }
      />
    ).dive();

    expect(component).toMatchSnapshot();
  });
});
