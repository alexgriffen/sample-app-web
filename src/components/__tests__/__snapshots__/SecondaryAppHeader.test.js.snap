// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Secondary App Header should render a full option component correctly 1`] = `
<WithStyles(AppBar)
  className="SecondaryAppHeader-root-1"
>
  <WithStyles(Toolbar)
    className="SecondaryAppHeader-toolBar-2"
  >
    <span
      className="SecondaryAppHeader-title-3"
    >
      Full options
    </span>
    <img
      className="SecondaryAppHeader-headerBot-4"
      src="test-file-stub"
      srcSet="test-file-stub"
    />
    <div
      className="SecondaryAppHeader-leftComponent-5"
    >
      <div />
    </div>
    <div
      className="SecondaryAppHeader-rightComponent-6"
    >
      <div />
    </div>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;

exports[`Secondary App Header should render the component correctly 1`] = `
<WithStyles(AppBar)
  className="SecondaryAppHeader-root-1"
>
  <WithStyles(Toolbar)
    className="SecondaryAppHeader-toolBar-2"
  >
    <Component />
    <Component />
    <Component />
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
